### DISCUSSION TITLE
SPAM - and the future of Keycloak community communication

### DISCUSSION THREAD

**Discussion by xgp** [keycloak/keycloak#20861] [Ideas]
Cross posting from https://groups.google.com/g/keycloak-dev/c/hEXZWu0S4FQ/m/t7CVVP_ABAAJ which references the spam problem on the keycloak-user mailing list (which has recently been banned by Google Groups):

Hi all,

As much as I need offers for cloned credit cards and fake identities, I'm not excited that the amount of spam on the Keycloak mailing lists seems to be going up. Strange to me that Google Groups doesn't effectively filter this.

Rather than just whack-a-mole this specific problem, I was thinking it could be a good starting point for a discussion on community communication. It seems like there are many places to go to ask questions, discuss features and get support. Rather than a clear objective for each medium, they all seem to have overlapping goals. Furthermore, rather than clarifying and reducing them, they are proliferating (I just found out about the CNCF Slack!). This presents several problems for community members:

For end users (especially newbies trying to get successful for the first time with Keycloak):
1. It is difficult to understand where to ask questions of a specific nature.
2. Their questions are often ignored, unacknowledged and not answered.
3. People end up cross-posting, hijacking threads, and other such measures to try to get an answer.
4. They can get quickly turned off from Keycloak.

For maintainers, and others who have experience with Keycloak (the "Answerers", for lack of a better term):
1. Having to visit 5+ places every day makes it difficult to allocate time and attention to issues.
2. It's not clear how a high-fidelity answer will be reused by the community, so it doesn't have to be answered again.
3. They end up answering the same question in many different places.
4. It can be frustrating to engage in an issue or conversation in one place, only to discover a "decision" has been discussed and decided in another.

Right now, I see at least 6 places where both types of community members are engaging. I've added some observations about each tool (my opinions):
1. Slack. Real-time, but ephemeral. Good for people hoping to get satisfaction quickly, but bad for preserving content for future discovery and use.
2. Discourse. Not as real-time as Slack, but lively discussions and issue handling. Great tool for community "answerers" to manage discussions, but rarely used by the maintainers. Excellent SEO and search, so this is the primary "front door" where end users come with issues.
3. Github Discussions. Overlap in functionality with Discourse, but bad SEO and search. Hard tool for power users to manage what has been read/answered. Seems to be the place where the maintainers present decisions on future roadmap and functionality.
4. Github Issues. Largely serves its purpose, but hard to navigate and find issues related to your own. Difficult to understand the roadmap and what will be in/out of future releases.
5. Mailing lists (Google Groups) - Somewhat stale and rarely used. List archives < Quarkus switch produce a lot of user confusion. Good SEO, but for old issues. Growth of spam.
6. Stack Overflow - Not controlled by the maintainers, but often a source of valuable, SEO'ed insights.

My core questions: Is it possible to downsize the number of tools in an attempt to solve the above problems? In addition to removing/centralizing tools, are there other tactics that can be used in order to serve the end user community better?

Thanks!
Garth

**Comment by nxadm** [comment]
Add #keycloak on LiberaChat IRC :)

**Reply by jonkoops** [reply]
The one thing we really don't need as a project is more places to divide our already distributed attention. We have too many channels of communication as it is.

**Reply by nxadm** [reply]
Just stating what already exists today.

**Reply by xgp** [reply]
Completely agree that there should be **fewer** tools, not more. My suggestion would be to drop 3 and 5 (and any others that come up), and just keep Discourse, Github Issues and Slack. I make a case for [Discourse over Github Discussions here](https://groups.google.com/g/keycloak-dev/c/hEXZWu0S4FQ/m/lyK1ev_CBAAJ).

**Comment by abstractj** [comment]
For Discourse, please check https://github.com/keycloak/keycloak/discussions/21112

**Comment by xgp** [comment]
Just following up here with the observation that the mailing list is (again) overrun with SPAM, and there are almost zero maintainer responses there and in the `keycloak` and `keycloak-dev` Slack channels. If the goal of shutting down the Discourse was to free maintainer time to focus on other channels, it has failed. Why not also shut down everything except GitHub Issues and Discussions? Given that there's basically nobody elsewhere, and suggesting otherwise is setting up new participants for failure, it seems like a better strategy to come where they can at least get some acknowledgement.
